auction-domain/test/lib/auction/domain/auction_test.rb

```diff
@@ -0,0 +1,47 @@
+require 'test_helper'
+require 'auction/domain/auction'
+require 'auction/domain/item'
+
+describe Auction do
+  it 'requires a deadline' do
+    proc { Auction.new }.must_raise Errors::InvalidDeadlineError
+  end
+
+  it 'requires a list of items' do
+    proc { Auction.new(deadline: Time.new(2015, 01, 01)) }.must_raise Errors::EmptyAuctionItemsError
+  end
+
+  describe '#items' do
+    it 'returns an immutable list of auction\'s items' do
+      auction = Auction.new(deadline: Time.new(2015, 01, 01), items: [Item.new('name', 1.0)])
```

Comment:
For you, is it ok to explicitly build the `Item` dependency? Is worth using doubles?